src/server/server_family.cc

```diff
@@ -1300,25 +1300,41 @@ GenericError ServerFamily::DoSave(bool new_version, string_view basename, Transa
                         StrCat(GlobalStateName(state), " - can not save database")};
   }
 
+  auto set_last_save_error = [this](const auto& save_info) mutable {
+    last_save_info_.last_error = save_info.error;
+    last_save_info_.last_error_time = save_info.save_time;
+    last_save_info_.failed_duration_sec = save_info.duration_sec;
+  };
+
   {
     std::lock_guard lk(save_mu_);
     if (save_controller_) {
       return GenericError{make_error_code(errc::operation_in_progress),
                           "SAVING - can not save database"};
     }
+
     save_controller_ = make_unique<SaveStagesController>(detail::SaveStagesInputs{
         new_version, basename, trans, &service_, fq_threadpool_.get(), snapshot_storage_});
+
+    auto res = save_controller_->InitResourcesAndStart();
+
+    if (res) {
+      CHECK_EQ(res->error, true);
```

Comment:
if this invariant won't hold will we reach UB? I do not think so.
If not, we should replace CHECK with DCHECK - to reduce the risk of crashing because of minor bugs.

Reply:
made it a DCHECK